src/adapters/MeetingsJSONAdapter.js

```diff
@@ -854,4 +885,59 @@ export default class MeetingsJSONAdapter extends MeetingsAdapter {
 
     return concat(initialState$, settingsEvent$);
   }
+
+  /**
+   * Returns an observable that emits the display data of the video options.
+   *
+   * @param {string} ID ID of the meeting for which to update display
+   * @returns {Observable.<MeetingControlDisplay>} Observable that emits control display data
+   * @private
+   */
+  videoControl(ID) {
+    const meeting = this.datasource[ID];
+    const videoDevices = defer(() => this.getDevices('videoinput'));
+    let initialState$;
+
+    if (meeting) {
+      initialState$ = videoDevices.pipe(
+        map((devices) => ({
+          ID: SWITCH_VIDEO_DEVICE,
+          tooltip: 'Video Devices',
+          state: MeetingControlState.INACTIVE,
+          options: devices,
+          selected: meeting.videoDeviceID,
+        })),
+      );
+    } else {
+      initialState$ = throwError(new Error(`Could not find meeting with ID "${ID}" to add video control`));
+    }
+
+    const videoEvent$ = fromEvent(document, SWITCH_VIDEO_DEVICE).pipe(
+      filter((event) => event.detail.ID === ID),
+      switchMap((event) => initialState$.pipe(
+        map((control) => (
+          {
+            ...control,
+            selected: event.detail.videoDeviceID,
+            state: MeetingControlState.ACTIVE,
+          })),
+      )),
+    );
+
+    return concat(initialState$, videoEvent$);
+  }
+
+  /**
+   * Returns an observable that emits the display data of a video control.
+   *
+   * @returns {Observable.<MeetingControlDisplay>} Observable that emits control display data
+   * @private
+   */
+  async switchVideo(ID, deviceId) {
+    const meeting = this.datasource[ID];
+
+    meeting.videoDeviceID = deviceId;
+    meeting.localVideo = await this.getStream({video: {deviceId: {exact: deviceId}}});
```

Comment:
you can set `meeting.videoDeviceID` after the `await` so that if the await fails, the id is not set either.